### PR TITLE
Clean code during investigation on JBDS-4227

### PR DESCRIPTION
**Pull request by apupier**
- remove useless check for non-nullity as variable already used
previously
- use constants instead of hardcoded and duplicated literals
- leverage genericity
- remove useless assignment
- invert equals to avoid potential NPEs
- use method reference instead of anonymous inner class

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
looks good to me, @robstryker, dare to review?

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/VagrantPoller.java [hunk 1]
```diff
@@ -70,12 +70,7 @@ public void beginPolling(IServer server, boolean expectedState) throws PollingEx
 		launchThread();
 	}
 	protected void launchThread() {
-		Thread t = new Thread(new Runnable(){
-			@Override
-			public void run() {
-				pollerRun();
-			}
-		}, "CDK Poller"); //$NON-NLS-1$
+		Thread t = new Thread(this::pollerRun, "CDK Poller"); //$NON-NLS-1$
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/VagrantPoller.java]: why not use a job?

**Comment by apupier** [comment]
> But given the very long methods and all the for's with nested if/else etc. I bet that there's plenty of room for another simple cleanup: split stuff up into smaller more concise methods.

Agree. I didn't take time to understood the code well enough to split into smaller methods. Unless I need to go deeper to resolve my issue JBDS-4227 I won't spend time for that.

**Comment by adietish** [comment]
@apupier ok, thanks! I'll add further changes according to the issues reported in Sonar.

**Comment by adietish** [comment]
added my changes on top #1384

**Pull request closed without merging**